shotover/src/transforms/kafka/sink_cluster/split.rs

```diff
@@ -0,0 +1,113 @@
+use super::KafkaSinkCluster;
+use crate::{
+    frame::{
+        kafka::{KafkaFrame, RequestBody},
+        Frame,
+    },
+    message::Message,
+};
+use kafka_protocol::{
+    indexmap::IndexMap,
+    messages::{
+        add_partitions_to_txn_request::AddPartitionsToTxnTransaction,
+        list_offsets_request::ListOffsetsTopic, produce_request::TopicProduceData,
+        AddPartitionsToTxnRequest, BrokerId, ListOffsetsRequest, ProduceRequest, TopicName,
+        TransactionalId,
+    },
+};
+use std::collections::HashMap;
+
+pub trait RequestSplitAndRouter {
+    type SubRequest;
```

Comment:
Should `SubRequest` be plural instead as it's IndexMap/Vec?

Reply:
yeah, that makes sense to me, I've changed it.